server/dataone_upload.py

```diff
@@ -242,6 +247,69 @@ def filter_items(item_ids, user):
     return {'dataone': dataone_objects, 'remote': remote_objects, 'local': local_objects}
 
 
+def create_upload_file_paths(item_ids, client):
+    """
+    Creates a file that lists the path that each item is located at. This is needed
+     to preserve the file structure when round tripping a tale.
+    :param item_ids: A list of items that are in the tale
+    :param client: The client object that is interfacing the member node
+    :type item_ids: list
+    :type client: MemberNodeClient_2_0
+    :return: The pid that describes the file in DataONE and its size
+    :rtype: tuple
+    """
+
+    """
+    We'll use a dict structure to hold the file contents during creation for
+     convenience. It will eventually be dumped to a string.
+    """
+    path_file = dict()
+
+    """
+    parentsToRoot returns a list of dicts; some of the dictionaries contain
+     irrelevant information that we can discard. If there comes a time where
+     we want more or less information, it can be done by adding removing
+     keys from the list.
+    """
+    remove_keys = ['created', 'updated', 'public', '_id', '_accessLevel',
+                   'creatorId', 'parentId', 'parentCollection', 'baseParentId',
+                   'baseParentType', 'access', 'lowerName', 'size']
+    # Get an admin user to access the folder
+    admin_user = ModelImporter.model('user').getAdmins()[0]
+
+    for item_id in item_ids:
+        item = ModelImporter.model('item').load(item_id,
+                                                level=AccessType.READ,
+                                                user=admin_user)
+        path = ModelImporter.model('item').parentsToRoot(item, force=True)
```

Comment:
If `path` is all you need, you could use `from girder.utility.path import getResourcePath` and then `path = getResourcePath('item', item, force=True)`

Reply:
Nice catch-I just made the changes and the output looks a lot cleaner